From 0f369229e271759af3affe32e9909107c5db4109 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Fri, 10 Jul 2015 15:46:04 +0800
Subject: [PATCH] MLK-11235-5 usb: chipidea: imx: add HSIC support for imx7d

Add HSIC support for imx7d. We have not supported HSIC as system
wakeup as well as HSIC remote wakeup function at DSM mode, since
the 24M OSC can't be off and the SoC internal regulators can't be
off at this mode, that will keep power consumption much higher.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 drivers/usb/chipidea/ci_hdrc_imx.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/chipidea/ci_hdrc_imx.c b/drivers/usb/chipidea/ci_hdrc_imx.c
index 108ac54..f78b3c1 100644
--- a/drivers/usb/chipidea/ci_hdrc_imx.c
+++ b/drivers/usb/chipidea/ci_hdrc_imx.c
@@ -139,10 +139,15 @@ static inline bool is_imx6sx_con(struct ci_hdrc_imx_data *imx_data)
 	return imx_data->data == &imx6sx_usb_data;
 }
 
+static inline bool is_imx7d_con(struct ci_hdrc_imx_data *imx_data)
+{
+	return imx_data->data == &imx7d_usb_data;
+}
+
 static inline bool imx_has_hsic_con(struct ci_hdrc_imx_data *imx_data)
 {
 	return is_imx6q_con(imx_data) ||  is_imx6sl_con(imx_data)
-		|| is_imx6sx_con(imx_data);
+		|| is_imx6sx_con(imx_data) || is_imx7d_con(imx_data);
 }
 
 /* Common functions shared by usbmisc drivers */
-- 
1.8.0

